src/sentry/options/manager.py

```diff
@@ -139,16 +188,32 @@ def delete(self, key):
         >>> from sentry import options
         >>> options.delete('option')
         """
-        cache_key = self._make_cache_key(key)
+        opt = self.lookup_key(key)
+
+        # If an option isn't able to exist in the store, we can't set it at runtime
+        assert not (opt.flags & self.FLAG_NOSTORE), '%r cannot be changed at runtime' % key
+        # Enforce immutability on key
+        assert not (opt.flags & self.FLAG_IMMUTABLE), '%r cannot be changed at runtime' % key
 
         Option.objects.filter(key=key).delete()
 
         try:
-            self.cache.delete(cache_key)
+            self.cache.delete(opt.cache_key)
+            return True
         except Exception:
             self.logger.warn(CACHE_UPDATE_ERR, key, exc_info=True)
+            return False
 
-    def update_cached_value(self, key, value):
-        cache_key = self._make_cache_key(key)
-
+    def update_cached_value(self, cache_key, value):
         self.cache.set(cache_key, value, self.ttl)
+
+    def register(self, key, default='', type=basestring, flags=FLAG_DEFAULT):
+        assert key not in self.registry, 'Option already registered: %r' % key
```

Comment:
It's probably a good idea to assert that the `default` is an instance of `type` to avoid shooting ourselves in the foot here.